### ISSUE TITLE
Inconsistent conditional result types

### ISSUE THREAD

**Issue by julian-perge** [terraform-aws-modules/terraform-aws-eks#2797]
## Description

Introduced from this commit: https://github.com/terraform-aws-modules/terraform-aws-eks/commit/bd4bda266e23635c7ca09b6e9d307b29ef6b8579

```
╷
│ Error: Inconsistent conditional result types
│ 
│   on .terraform/modules/eks.eks/modules/eks-managed-node-group/main.tf line 179, in resource "aws_launch_template" "this":
│  179:     for_each = length(var.license_specifications) > 0 ? var.license_specifications : []
│     ├────────────────
│     │ var.license_specifications is object with no attributes
│ 
│ The true and false result expressions must have consistent types. The
│ 'true' value is object, but the 'false' value is tuple.
╵
```

## Versions

- Module version [Required]: `19.17.3`

- Terraform version: `1.6.2`
- Provider version(s): `hashicorp/aws v4.67.0`

- Don't provide a `license_specification` variable

**Comment by samsonquantifi** [comment]
we ran into this as well. temporary work around is to override the license_specification to return an empty array:

We're using eks managed node_groups, so this is an example.
```
  eks_managed_node_group_defaults = {
    license_specifications = []
  }
  ```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 19.17.4 :tada:
